pkg/apis/resource/types.go

```diff
@@ -128,13 +128,39 @@ type ResourceSliceSpec struct {
 	// +oneOf=NodeSelection
 	AllNodes bool
 
-	// Devices lists some or all of the devices in this pool.
+	// Devices lists some or all of the devices in this pool. Devices
+	// listed here do not inherit shared attributes from the slice.
 	//
 	// Must not have more than 128 entries.
 	//
 	// +optional
 	// +listType=atomic
 	Devices []Device
+
+	// SharedCapacity defines the set of shared capacity consumable by
+	// partitionable devices in this ResourceSlice.
+	//
+	// Must not have more than 128 entries. Names must be C identifiers
+	// that are unique inside the ResourceSlice.
+	//
+	// +optional
+	SharedCapacity map[string]resource.Quantity
```

Comment:
I don't think we should include this directly in the `Device` type anymore. As mentioned in my comment below, I think we should do this in a separate object. I will try and find time to spec out the proposal in more detail this week.

Reply:
Correct. I picked option 3 partly arbitrarily, partly because it contained two different features that we might want to add.